services/api/src/admin_new_intersection.py

```diff
@@ -181,18 +209,27 @@ def options(self):
         # CORS support
         return ("", 204, self.options_headers)
 
-    def get(self):
+    @require_permission(
+        required_role=ORG_ROLE_LITERAL.USER,
+    )
+    def get(self, permission_result: PermissionResult):
         logging.debug("AdminNewIntersection GET requested")
-        return (get_allowed_selections(), 200, self.headers)
 
-    def post(self):
+        return (get_allowed_selections(permission_result.user), 200, self.headers)
+
+    @require_permission(
+        required_role=ORG_ROLE_LITERAL.OPERATOR,
+    )
+    def post(self, permission_result: PermissionResult):
         logging.debug("AdminNewIntersection POST requested")
         # Check for main body values
         schema = AdminNewIntersectionSchema()
         errors = schema.validate(request.json)
         if errors:
             logging.error(str(errors))
             abort(400, str(errors))
+        enforce_add_intersection_org_permissions(
+            permission_result.user, permission_result.qualified_orgs, request.json
+        )
 
-        data, code = add_intersection(request.json)
-        return (data, code, self.headers)
+        return (add_intersection(request.json), 200, self.headers)
```

Comment:
Do we want to catch errors in our exposed methods to return appropriate codes and messages? Currently if the user fails a permissions check it looks like we'd just get a 500 internal server error from the error raise. But a better return would be a 401 unauthorized

Reply:
Looks like I'm mistaken on this one with the werkzeug exceptions. I hadn't seen that before, it's an interesting handling